Source/Scene/ImplicitTileMetadata.js

```diff
@@ -0,0 +1,152 @@
+ * @param {MetadataClass} [options.class] The class that group metadata conforms to.
+ *
+ * @alias ImplicitTileMetadata
+ * @constructor
+ *
+ * @private
+ */
+export default function ImplicitTileMetadata(options) {
+  options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+
+  //>>includeStart('debug', pragmas.debug);
+  Check.typeOf.object("options.implicitSubtree", options.implicitSubtree);
+  Check.typeOf.object(
+    "options.implicitCoordinates",
+    options.implicitCoordinates
+  );
+  //>>includeEnd('debug');
+
+  this._class = options.class;
+
+  var subtree = options.implicitSubtree;
+  this._metadataTable = subtree.metadataTable;
+  this._entityId = subtree.getEntityId(options.implicitCoordinates);
+
+  var subtreeExtension = subtree.metadataExtension;
+  this._extensions = subtreeExtension.extensions;
+  this._extras = subtreeExtension.extras;
+}
+
+Object.defineProperties(ImplicitTileMetadata.prototype, {
+  /**
+   * The class that properties conform to.
+   *
+   * @memberof ImplicitTileMetadata.prototype
+   * @type {MetadataClass}
+   * @readonly
+   */
+  class: {
+    get: function () {
+      return this._class;
+    },
+  },
+
+  /**
+   * Extras in the JSON object.
+   *
+   * @memberof MetadataGroup.prototype
+   * @type {*}
+   * @readonly
+   */
+  extras: {
+    get: function () {
+      return this._extras;
+    },
+  },
+
+  /**
+   * Extensions in the JSON object.
+   *
+   * @memberof MetadataGroup.prototype
+   * @type {Object}
+   * @readonly
+   */
+  extensions: {
+    get: function () {
+      return this._extensions;
+    },
+  },
+});
+
+/**
+ * Returns whether this property exists.
+ *
+ * @param {String} propertyId The case-sensitive ID of the property.
+ * @returns {Boolean} Whether this property exists.
+ */
+ImplicitTileMetadata.prototype.hasProperty = function (propertyId) {
+  return this._metadataTable.hasProperty(propertyId);
+};
+
+/**
+ * Returns an array of property IDs.
+ *
+ * @param {String[]} [results] An array into which to store the results.
+ * @returns {String[]} The property IDs.
+ */
+ImplicitTileMetadata.prototype.getPropertyIds = function (results) {
+  return this._metadataTable.getPropertyIds(results);
+};
+
+/**
+ * Returns a copy of the value of the property with the given ID.
+ * <p>
+ * If the property is normalized the normalized value is returned.
+ * </p>
+ *
+ * @param {String} propertyId The case-sensitive ID of the property.
+ * @returns {*} The value of the property or <code>undefined</code> if the property does not exist.
+ */
+ImplicitTileMetadata.prototype.getProperty = function (propertyId) {
+  return this._metadataTable.getProperty(this._entityId, propertyId);
+};
+
+/**
+ * Sets the value of the property with the given ID.
+ * <p>
+ * If the property is normalized a normalized value must be provided to this function.
+ * </p>
+ * <p>
+ * If a property with the given ID doesn't exist, it is created.
```

Comment:
> If a property with the given ID doesn't exist, it is created

This applies to `TilesetMetadata` and `GroupMetadata` but doesn't apply to anything that uses metadata tables. Maybe we should remove this capability from everywhere until we figure out a consistent solution.

Also we're a bit inconsistent about the level of detail in the doc: `MetadataTable.prototype.setProperty` has more detail than `TilesetMetadata`/`GroupMetadata`.

In general we should be more consistent with these different types of metadata. Most of that is on me from the initial set of PRs.